src/main/java/net/wasdev/wlp/common/plugins/util/InstallFeatureUtil.java

```diff
@@ -0,0 +1,559 @@
+        }
+        if (hasUnsupportedParameters(from, to, pluginListedEsas)) {
+            throw new PluginScenarioException("Cannot install features from a Maven repository when using the 'to' or 'from' parameters or when specifying ESA files.");
+        }
+    }
+    
+    /**
+     * Log debug
+     * @param msg
+     */
+    public abstract void debug(String msg);
+
+    /**
+     * Log debug
+     * @param msg
+     * @param e
+     */
+    public abstract void debug(String msg, Throwable e);
+
+    /**
+     * Log debug
+     * @param e
+     */
+    public abstract void debug(Throwable e);
+
+    /**
+     * Log warning
+     * @param msg
+     */
+    public abstract void warn(String msg);
+
+    /**
+     * Log info
+     * @param msg
+     */
+    public abstract void info(String msg);
+
+    /**
+     * Returns whether debug is enabled by the current logger
+     * @return whether debug is enabled
+     */
+    public abstract boolean isDebugEnabled();
+
+    /**
+     * Download the artifact from the specified Maven coordinates, or retrieve it from the cache if it already exists.
+     * 
+     * @param groupId The group ID
+     * @param artifactId The artifact ID
+     * @param type The type e.g. esa
+     * @param version The version
+     * @return The file corresponding to the downloaded artifact
+     * @throws PluginExecutionException If the artifact could not be downloaded
+     */
+    public abstract File downloadArtifact(String groupId, String artifactId, String type, String version)
+            throws PluginExecutionException;
+    
+    /**
+     * Combine the given String collections into a set
+     * 
+     * @param collections a collection of strings
+     * @return the combined set of strings
+     */
+    @SafeVarargs
+    public static Set<String> combineToSet(Collection<String>... collections) {
+        Set<String> result = new HashSet<String>();
+        for (Collection<String> collection : collections) {
+            result.addAll(collection);
+        }
+        return result;
+    }
+    
+    /**
+     * Get the set of features defined in the server.xml if there were no features listed in the plugin configuration
+     * @param serverDirectory The server directory containing the server.xml
+     * @param noPluginListedFeatures true if there were no features listed in the plugin configuration
+     * @return the set of features that should be installed from server.xml
+     */
+    public Set<String> getServerFeatures(File serverDirectory, boolean noPluginListedFeatures) {
+        // parse server.xml features only if there are no configured features in the pom
+        Set<String> result = new HashSet<String>();
+        if (noPluginListedFeatures) {
+            debug("No features were listed for the plugin. Using server.xml.");
+            result.addAll(getConfigDropinFeatures(serverDirectory, "overrides"));
```

Comment:
Shouldn't this order be reversed (parse `overrides` after `defaults`) since the item that is parsed last will overwrite previous items?